### PR TITLE
Test the interaction of document.domain and iframe removal

### PR DESCRIPTION
**Pull request by domenic**
See https://crbug.com/1095145.

### CODE REVIEW

File: html/browsers/windows/document-domain-removed-iframe.html [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+promise_test(t => {
+  return new Promise(resolve => {
+    const iframe = document.createElement("iframe");
+    iframe.onload = t.step_func(() => {
+      document.domain = document.domain;
+      const { contentWindow } = iframe;
+      assert_equals(contentWindow.status, "");
+      resolve();
+    });
+    iframe.src = "resources/document-domain-setter.html";
```
Comment (annevk) [inline html/browsers/windows/document-domain-removed-iframe.html:52]: This part is required? Should there be another test where only the parent sets it?

File: html/browsers/windows/document-domain-removed-iframe.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+<script>
+"use strict";
+
+promise_test(t => {
+  return new Promise(resolve => {
+    const iframe = document.createElement("iframe");
+    iframe.onload = t.step_func(() => {
+      const { contentWindow } = iframe;
+      assert_equals(contentWindow.status, "");
```
Comment (annevk) [inline html/browsers/windows/document-domain-removed-iframe.html:21]: Should we assert more than status?

File: html/browsers/windows/document-domain-removed-iframe.html (hunk 1 above)
Comment (domenic) [reply html/browsers/windows/document-domain-removed-iframe.html:52]: Maybe? I don't know; I didn't really want to sign up for writing a lot of tests here, I just had some regression tests in a crbug and I thought it'd be a good idea to contribute them to web platform tests too.

File: html/browsers/windows/document-domain-removed-iframe.html (hunk 2 above)
Comment (domenic) [reply html/browsers/windows/document-domain-removed-iframe.html:21]: Status suffices to reproduce the Chrome bug.

**Review by annevk — APPROVED** [review]

**Pull request merged**